config_generator/src/main.rs

```diff
@@ -0,0 +1,191 @@
+    let exsiting_id = namespaces.into_iter().find_map(|namespace| {
+        if namespace.title == "sxg-OCSP" {
+            Some(namespace.id)
+        } else {
+            None
+        }
+    });
+    if let Some(id) = exsiting_id {
+        return id;
+    }
+    let namespace = wrangler::kv::namespace::create(&client, account_id, "OCSP")
+        .unwrap()
+        .result;
+    namespace.id
+}
+
+fn read_certificate_pem_file(path: &str) -> Result<String, String> {
+    let text =
+        std::fs::read_to_string(path).map_err(|_| format!(r#"Failed to read file "{}""#, path))?;
+    let certs = pem::parse_many(&text);
+    if certs.len() == 1 && certs[0].tag == "CERTIFICATE" {
+        Ok(text)
+    } else {
+        Err(format!(r#"File "{}" is not a valid certificate PEM"#, path))
+    }
+}
+
+const CONFIG_FILE: &'static str = "cloudflare_worker/wrangler.toml";
+const CONFIG_EXAMPLE_FILE: &'static str = "cloudflare_worker/wrangler.example.toml";
+
+// Read and parse `wrangler.toml`.
+// `wrangler.example.toml` will be read if `wrangler.toml` does not exist.
+// This function panics if `wrangler.toml` contains syntax error,
+// even when a valid `wrangler.example.toml` exists.
+fn read_exsiting_config() -> (WranglerConfig, SxgConfig) {
+    let wrangler_config = std::fs::read_to_string(CONFIG_FILE)
+        .or_else(|_| std::fs::read_to_string(CONFIG_EXAMPLE_FILE))
+        .unwrap();
+    let wrangler_config: WranglerConfig = toml::from_str(&wrangler_config).unwrap();
+    let sxg_config: SxgConfig = serde_yaml::from_str(&wrangler_config.vars.sxg_config).unwrap();
+    (wrangler_config, sxg_config)
+}
+
+fn main() -> Result<(), std::io::Error> {
+    goto_repository_root()?;
+    let (mut wrangler_config, mut sxg_config) = read_exsiting_config();
+    loop {
+        let cert = read_certificate_pem_file("credentials/cert.pem");
+        let issuer = read_certificate_pem_file("credentials/issuer.pem");
+        if cert.is_ok() && issuer.is_ok() {
+            println!("Succesffuly read certificates");
+            wrangler_config.vars.cert_pem = cert.unwrap();
+            wrangler_config.vars.issuer_pem = issuer.unwrap();
+            break;
+        }
+        if let Err(msg) = cert {
+            println!("{}", msg);
+        }
+        if let Err(msg) = issuer {
+            println!("{}", msg);
+        }
+        println!(
+            "To generate certificates, please refer to\n\
+            \thttps://github.com/google/sxg-rs/blob/main/credentials/README.md"
+        );
+        Input::<String>::new()
+            .with_prompt("Press ENTER key after you generate the certificates")
+            .allow_empty(true)
+            .interact_text()
+            .unwrap();
+    }
+    let user = get_global_uesr();
+    wrangler_config.account_id = Input::new()
+        .with_prompt("What's your Cloudflare account ID?")
+        .with_initial_text(wrangler_config.account_id)
```

Comment:
When re-running every ~90 days for the new certificate, it would be frustrating to have to hit `<Enter><Enter><Enter><Enter>` to confirm all the defaults. Can you change this so it doesn't prompt if there's an existing valid value? (Ditto all fields below.)

Reply:
Or, optionally, leave the current behavior the same, but add a `-y` flag that accepts all defaults, and exits with a nonzero code if there is no default (so it doesn't hang if run from a crontab).

Reply:
The reason we can't skip the prompt in this PR is because the `wrangler.config.toml` has all fields filled.
 I think in the future, we can remove the example toml, so that the code will know whether this is the initial run. At that time we can add the feature to skip the existing field.

Reply:
Perhaps we can skip the prompt when we know that the config was loaded successfully from CONFIG_FILE (not CONFIG_EXAMPLE_FILE). What do you think?

Reply:
I created issue #34, to set `config-generator` as non-interactive, and we will no longer have any prompt.